stan/math/rev/fun/falling_factorial.hpp

```diff
@@ -12,11 +12,34 @@ namespace math {
 inline var falling_factorial(const var& a, int b) {
   auto digamma_ab = digamma(a.val() + 1) - digamma(a.val() - b + 1);
   return make_callback_var(falling_factorial(a.val(), b),
-                           [a, b, digamma_ab](auto& vi) mutable {
+                           [a, digamma_ab](auto& vi) mutable {
                              a.adj() += vi.adj() * vi.val() * digamma_ab;
                            });
 }
 
+template <typename T, require_eigen_t<T>* = nullptr>
+inline auto falling_factorial(const var_value<T>& a, int b) {
+  auto digamma_ab = to_arena(digamma(a.val().array() + 1)
+                             - digamma(a.val().array() - b + 1));
+  return make_callback_var(
+      falling_factorial(a.val(), b), [a, digamma_ab](auto& vi) mutable {
+        a.adj().array() += vi.adj().array() * vi.val().array() * digamma_ab;
+      });
+}
+
+template <typename T, typename StdVec, require_eigen_t<T>* = nullptr,
+          require_vector_like_vt<std::is_integral, StdVec>* = nullptr>
```

Comment:
I noticed the other functions use `std::vector<int>` rather the `require_`, any reason for the preference here?

Reply:
Honestly idt so, but this is all cleaned up with your suggestions above